lib/log/log.go

```diff
@@ -112,3 +113,16 @@ func Stderr(ctx context.Context) context.Context {
 
 	return With(ctx, l)
 }
+
+// WithTimeout returns context.WithTimeout(ctx, timeout) but timeout is overridden with D2_TIMEOUT if set
+func WithTimeout(ctx context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
This seems like an odd place to have this function as it is not related to `log` at all. If this is not a convention, I'd say it should be somewhere else, like `main.go` is fine

Reply:
hmm I'll add a `lib/util.go` file, I don't think we can be importing from main.go in other libraries

Reply:
nah make a `lib/time`